### ISSUE TITLE
semver crate parsing feature parity with node-semver

### ISSUE THREAD

**Issue by zkat** [entropic-dev/ds#37]
Currently, there's a number of open issues in the semver crate that involve bringing it to feature parity with node-semver. We need this because of the sorts of things we'll run into in the wild. This issue is a tracking issue for that work -- it's likely we'll need to do it ourselves if we want it done because dev on the crate seems stalled.

Off the top of my head, these are the missing features:

1. "loose" parsing, which involves handling garbage in semver strings. Because it's actually needed. This is an extra option in node-semver but it's used throughout NPM and the ecosystem has come to rely on it.
2. Space-separated range specifiers. This is supported by semver-parser but semver hasn't been updated to use it.
3. Or (`||`)-separated ranges
4. Other stuff?

**Comment by larsgw** [comment]
What's the time frame? I'd love to help but I don't know if I can fit it into my schedule today.

**Comment by zkat** [comment]
Yay Lars! Hi!

The time frame is "over the next week or two". Probably longer but I'd rather get it done sooner rather than later. Does that work for you?

**Comment by zkat** [comment]
Update: it looks like Steve has a full rewrite that just needs some help getting over the finish line: https://twitter.com/steveklabnik/status/1203691014055190528?s=19

**Comment by larsgw** [comment]
Ah, cool.

**Comment by zkat** [comment]
You should definitely ping him about it if you're still interested. It seems like he could still use a bit of help :)

**Comment by zkat** [comment]
Heyo! Did y'all manage to get in touch and get the ball rolling? Any updates?

**Comment by larsgw** [comment]
Yep, I made a spreadsheet to track progress but there's not much of any yet, unfortunately.

**Comment by zkat** [comment]
Thanks for the update! 👍🏼👍🏼👍🏼

**Comment by larsgw** [comment]
I started working on it a week ago but know I'm waiting for Steve to answer some minor but progress-blocking questions (for me at least), and I think he may be out during the holidays.

**Comment by larsgw** [comment]
Current state (some of which I said on discord earlier):

  - I started working on the main missing features, most of which are done now;
  - Loose parsing, `clean()` and `coerce()` (as from `node-semver`) are still pending feedback;
  - There are some other things left to do that I did not think of earlier, namely
    - documentation,
    - proper use of `Error` (although the use of `String` may have been on purpose, will have to discuss) and
    - going through the existing GitHub issues (which is not as much as it may sound)

**Comment by zkat** [comment]
This is awesome! Thanks again for the update!
